docs(action-strip): Add topic for Action Strip (#1915)
IgniteUI/igniteui-docfx: en/components/action-strip.md

---
title: Action Strip
_description: The Action Strip represents a template area for one or more actions. The Action Strip requires to be inside a relative container, as it is going to overlay it.
_keywords: Ignite UI for Angular, UI controls, Angular widgets, web widgets, UI widgets, Angular, Native Angular Components Suite, Native Angular Controls, Native Angular Components Library, Angular ActionStrip components, Angular ActionStrip directives, Angular ActionStrip controls
---

## Action Strip
<p class="highlight">

> The Ignite UI for Angular [`igxActionStrip`]({environment:angularApiUrl}/classes/igxactionstripcomponent.html) represents a template area for one or more actions. The Action Strip is intended to provide additional UI and functionality for a specific element container. This container should be relative, because the Action Strip attempts to overlay it. This provides an easy way to position the actions over the parent. Even though the parent is overlapped by the Action Strip the main interactions and user access to the container will remain available.

Reviewer: ```suggestion
The Ignite UI for Angular [`igxActionStrip`]({environment:angularApiUrl}/classes/igxactionstripcomponent.html) represents a template area for one or more actions. The Action Strip is intended to provide additional UI and functionality for a specific target container. This container should be positioned relatively as the Action Strip attempts to overlay it and is itself positioned absolutely. This approach provides an easy way to position the elements of the action strip on top of the parent container. Even though the parent container is overlapped by the Action Strip, the parent will remain accessible.
```

</p>

<div class="divider--half"></div>

#### Demo

<div class="sample-container loading" style="height: 400px;">
    <iframe id="action-strip-paragraph-sample" frameborder="0" seamless width="100%" height="100%" src="{environment:demosBaseUrl}/menus/action-strip-paragraph" onload="onSampleIframeContentLoaded(this);"></iframe>
</div>
<div>
    <button data-localize="stackblitz" disabled class="stackblitz-btn" data-iframe-id="action-strip-paragraph-sample" data-demos-base-url="{environment:demosBaseUrl}">view on stackblitz</button>
</div>

<div class="divider--half"></div>

### Usage

> In order to initialize and position correctly the Action Strip it needs to be inside relative container:

Reviewer: ```suggestion
In order to initialize and position the Action Strip correctly, it needs to be inside a relatively positioned container:
```

```html

> <div style="position:relative;width:100px;height:100px;">

Reviewer: ```suggestion
<div style="position:relative; width:100px; height:100px;">
```

    <igx-action-strip>
        <button igxButton (click)="makeTextBold()">
            <igx-icon>format_bold</igx-icon>
        </button>
    </igx-action-strip>
<div>
```

By default the Action Strip will be visible, but this can be controlled by the [`hidden`]({environment:angularApiUrl}/classes/igxactionstripcomponent.html#hidden) @Input property.

> #### Menu look and feel

Reviewer: ```suggestion
#### Look and Feel
```

Author: This section is about the menu look and feel of the action strip. Any other idea for a header?

Reviewer 2: I think you can dismiss this suggestion

> For scenarios with more than three actions, which are overlapping the main content it might be a good idea to use [`IgxActionStripMenuItem`]({environment:angularApiUrl}/classes/igxactionstripmenuitemdirective.html) directive. Any item in an Action Strip marked with `*igxActionStripMenuItem` will be resulting in a three-dot button, which toggles a dropdown containing those items.

Reviewer: ```suggestion
For scenarios with more than three action items, it might be a good idea to use the [`IgxActionStripMenuItem`]({environment:angularApiUrl}/classes/igxactionstripmenuitemdirective.html) directive. When an item in the Action Strip is marked with the `*igxActionStripMenuItem` structural directive, the resulting Action Strip will include a three-dot button. Pressing this button toggles a dropdown containing all marked items.
```

```html
<div style="position:relative;width:100px;height:100px;">
    <igx-action-strip>
        <button *igxActionStripMenuItem igxButton (click)="alignTextLeft()">
            <igx-icon>format_align_left</igx-icon>
        </button>
        <button *igxActionStripMenuItem igxButton (click)="alignTextCenter()">
            <igx-icon>format_align_center</igx-icon>
        </button>
        <button *igxActionStripMenuItem igxButton (click)="alignTextRight()">
            <igx-icon>format_align_right</igx-icon>
        </button>
    </igx-action-strip>
</div>
```

<div class="sample-container loading" style="height: 400px;">
    <iframe id="action-strip-menu-sample" frameborder="0" seamless width="100%" height="100%" src="{environment:demosBaseUrl}/menus/action-strip-paragraph-menu" onload="onSampleIframeContentLoaded(this);"></iframe>
</div>
<div>
    <button data-localize="stackblitz" disabled class="stackblitz-btn" data-iframe-id="action-strip-menu-sample" data-demos-base-url="{environment:demosBaseUrl}">view on stackblitz</button>
</div>

#### Reusing the Action Strip

> Single Action Strip can be used for more than one element. As long as the actions of those elements are not required to be visible simultaneously.

Reviewer: ```suggestion
A single Action Strip instance can be used in multiple places in the document as long as the action items are not shown simultaneously.
```

> The Action Strip can change the container element it is shown for.

Reviewer: ```suggestion
The Action Strip can change its parent container.
```

> This is achieved by changing the [`context`]({environment:angularApiUrl}/classes/igxactionstripcomponent.html#context).

Reviewer: ```suggestion
This is achieved by changing the [`context`]({environment:angularApiUrl}/classes/igxactionstripcomponent.html#context) of the Action Strip.
```

> The best way to change the `context` is by using the [`show`]({environment:angularApiUrl}/classes/igxactionstripcomponent.html#show) API method and pass the `context` as parameter. The `context` should be an instance of a component and should have an accessible `element` property.

Reviewer: ```suggestion
The best way to change the `context` is by using the [`show`]({environment:angularApiUrl}/classes/igxactionstripcomponent.html#show) API method and passing the `context` as an argument. The `context` should be an instance of a component and should have an accessible `element` property of the `ElementRef` type.
```

> >Note: The `show` API method uses [Angular Renderer2](https://angular.io/api/core/Renderer2) to append the Action Strip into that `element` .

Reviewer: ```suggestion
>Note: The `show` API method uses [Angular Renderer2](https://angular.io/api/core/Renderer2) to append the Action Strip into that `element`.
```

### Usage in Grids

> Action strip provides functionality and UI for IgxGrid.

Reviewer: ```suggestion
The Action Strip provides additional functionality and UI for the IgxGrid.
```

> All that can be utilized with grid action components. 

Reviewer: ```suggestion
This additional functionality and UI is provided by two additional components. 
```

> We are providing two default grid action components:

Reviewer: Here's the list of components and what they do:

> - [`IgxGridEditingActionsComponent`]({environment:angularApiUrl}/classes/igxgrideditingactionscomponent.html) - includes functionality and UI related to grid editing. It allows starting edit mode for cell or row, depending on rowEditable option of the grid and deleting rows.

Reviewer: ```suggestion
- [`IgxGridEditingActionsComponent`]({environment:angularApiUrl}/classes/igxgrideditingactionscomponent.html) - includes functionality and UI elements related to grid editing. It allows you to quickly toggle edit mode for cells or rows, depending on the value of the `rowEditable` option of the grid and whether deleting rows is allowed.
```

> - [`IgxGridPinningActionsComponent`]({environment:angularApiUrl}/classes/igxgridpinningactionscomponent.html) - includes functionality and UI related to grid pinning. It allows pinning rows and navigating from a pinned row to the disabled row.

Reviewer: ```suggestion
- [`IgxGridPinningActionsComponent`]({environment:angularApiUrl}/classes/igxgridpinningactionscomponent.html) - includes functionality and UI elements related to grid pinning. It allows you to quickly pin rows and navigate between pinned and disabled rows.
```

        <ng-template igxCell let-cell="cell" let-val>
            <div class="cell-template" (mouseover)="actionStrip.show(cell.row)" (mouseout)="actionStrip.hide()">
                <span>{{val}}</span>
            </div>
        </ng-template>
    </igx-column>

    <igx-action-strip #actionStrip>
        <igx-grid-pinning-actions></igx-grid-pinning-actions>
        <igx-grid-editing-actions></igx-grid-editing-actions>
    </igx-action-strip>
</igx-grid>
```
>Note: These components inherit [`IgxGridActionsBaseDirective`]({environment:angularApiUrl}/classes/igxgridactionsbasedirective.html) and when creating a custom grid action component, it should also inherit `IgxGridActionsBaseDirective`.

<div class="sample-container loading" style="height: 600px;">
    <iframe id="action-strip-grid-sample" frameborder="0" seamless width="100%" height="100%" src="{environment:demosBaseUrl}/grid/grid-action-strip" onload="onSampleIframeContentLoaded(this);"></iframe>
</div>
<div>
    <button data-localize="stackblitz" disabled class="stackblitz-btn" data-iframe-id="action-strip-grid-sample" data-demos-base-url="{environment:demosBaseUrl}">view on stackblitz</button>
</div>

### Styling

> The Action Strip allows styling through the [Ignite UI for Angular Theme Library](../themes/component-themes.md). The Action Strip [theme]({environment:sassApiUrl}/index.html#function-igx-action-strip-theme) exposes properties, which allow the customization of the Action Strip.
> 
> In the below steps, we are going through the steps of customizing the Action Strip styling.
> 
> #### Importing global theme
> To begin the customization of the Action Strip, you need to import the `index` file, where all styling functions and mixins are located.

Reviewer 2: ```suggestion
To customize the Action Strip, you first need to import the `index` file, where all styling functions and mixins are located.
```

```scss
@import '~igniteui-angular/lib/core/styles/themes/index'
```

> In order to create a new theme `igx-action-strip-theme` should be extended:

Reviewer 2: ```suggestion
Next, we have to create a new theme that extends the `igx-action-strip-theme` and pass the parameters which we'd like to change:
```

```scss
$custom-strip: igx-action-strip-theme(
    $background: rgba(150, 133, 143, 0.4),
    $actions-background: rgba(109, 121, 147, 0.6),
    $color:  null,
    $delete-action: null,
    $actions-border-radius: 0
);
```

> #### Including themes
> The last step is to include the component theme in our application.

Reviewer 2: ```suggestion
The last step is to include the newly created component theme in our application.
```

> If `$legacy-support` is set to `false`(default), include the component css variables like that:

Reviewer 2: ```suggestion
When `$legacy-support` is set to `false`(default), include the component css variables like this:
```

```scss
@include igx-css-vars($custom-strip);
```

> If `$legacy-support` is set to `true`, include the component theme like that:

Reviewer 2: ```suggestion
When `$legacy-support` is set to `true`, include the component theme like this:
```

```scss
@include igx-action-strip($custom-strip);
```

<div class="sample-container loading" style="height: 400px;">
    <iframe id="action-strip-styling-sample" frameborder="0" seamless width="100%" height="100%" src="{environment:demosBaseUrl}/menus/action-strip-styling" onload="onSampleIframeContentLoaded(this);"></iframe>
</div>
<div>
    <button data-localize="stackblitz" disabled class="stackblitz-btn" data-iframe-id="action-strip-styling-sample" data-demos-base-url="{environment:demosBaseUrl}">view on stackblitz</button>
</div>

### API and Style References

For more detailed information regarding the Action Strip API, refer to the following links:
* [`IgxActionStripComponent API`]({environment:angularApiUrl}/classes/igxactionstripcomponent.html)

The following built-in CSS styles helped us achieve this Action Strip layout:

* [`IgxActionStripComponent Styles`]({environment:sassApiUrl}/index.html#function-igx-action-strip-theme)

> Additional components and/or directives that can be used within Action Strip:

Reviewer 2: ```suggestion
Additional components and/or directives that can be used within the Action Strip:
```
